pkg/clustermgmt/clusterWatch.go

```diff
@@ -0,0 +1,247 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package clustermgmt
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"sync"
+	"time"
+
+	clusterv1beta1 "github.com/stolostron/multicloud-operators-foundation/pkg/apis/internal.open-cluster-management.io/v1beta1"
+	"github.com/stolostron/search-indexer/pkg/config"
+	"github.com/stolostron/search-indexer/pkg/database"
+	"github.com/stolostron/search-indexer/pkg/model"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+var dynamicClient dynamic.Interface
+var dao database.DAO
+
+const managedClusterGVR = "managedclusters.v1.cluster.open-cluster-management.io"
+const managedClusterInfoGVR = "managedclusterinfos.v1beta1.internal.open-cluster-management.io"
+
+// Watches ManagedCluster objects and updates the database with a Cluster node.
+func WatchClusters() {
+	klog.Info("Begin ClusterWatch routine")
+
+	dynamicClient = config.GetDynamicClient()
+	dynamicFactory := dynamicinformer.NewDynamicSharedInformerFactory(dynamicClient, 60*time.Second)
+
+	// Create GVR for ManagedCluster and ManagedClusterInfo
+	managedClusterGvr, _ := schema.ParseResourceArg(managedClusterGVR)
+	managedClusterInfoGvr, _ := schema.ParseResourceArg(managedClusterInfoGVR)
+
+	//Create Informers for ManagedCluster and ManagedClusterInfo
+	managedClusterInformer := dynamicFactory.ForResource(*managedClusterGvr).Informer()
+	managedClusterInfoInformer := dynamicFactory.ForResource(*managedClusterInfoGvr).Informer()
+
+	// Create handlers for events
+	handlers := cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			klog.V(4).Info("clusterWatch: AddFunc")
+			processClusterUpsert(obj.(*unstructured.Unstructured).GetName())
```

Comment:
Pass the entire resource object to avoid having to do a Get() later in the transform.

Reply:
Done